src/test_servers.py

```diff
@@ -32,3 +44,38 @@ def test_response_error():
     output_ = 'HTTP/1.1 500 Internal Server Error'
     http_list = response_error().split('\r\n')
     assert http_list[0] == output_
+
+
+def test_parse_method():
+    """Assert for correct method."""
+    from server import parse_request
+    request = 'PUT / HTTP/1.1'
+    with pytest.raises(RuntimeError):
+        parse_request(request)
+
+
+def test_parse_protocol():
+    """Assert for correct method."""
+    from server import parse_request
+    request = 'GET / HTTP/2.1'
+    with pytest.raises(RuntimeError):
+        parse_request(request)
+
+
+def test_parse_uri():
+    """Assert for correct method."""
+    from server import parse_request
+    request = 'GET HTTP/1.1'
+    with pytest.raises(RuntimeError):
+        print(parse_request(request))
+
+
+def test_parse():
+    """Assert 404 error works correctly."""
+    from server import response_error
+    request = u'404 Page Not Found'
+    response = response_error(request)
+    response = response.split()
+    assert response[1] == '404'
+
+
```

Comment:
There doesn't seem to be a test checking the behavior of `parse_request()` when everything should be working perfectly.